base/ltxref.dtx

```diff
@@ -343,16 +383,29 @@
 %    to something like |\foo{4.d}|.  Change made 20 Jul 88.
 %
 %    \begin{macrocode}
+%</2ekernel>
+%<*2ekernel|latexrelease>
+%<latexrelease>\IncludeInRelease{2023/06/01}%
+%<latexrelease>                 {\label}{store five label arguments}%
+\providecommand\@currentlabelname{}
+\providecommand\@currentHref{}
+\providecommand\@kernel@currentdata{}
+\providecommand\label@name{}
+\NewHook{label}
 \def\label#1{\@bsphack
+  \begingroup
+  \def\label@name{#1}%
+  \UseHook{label}%
   \protected@write\@auxout{}%
-         {\string\newlabel{#1}{{\@currentlabel}{\thepage}}}%
+         {\string\newlabel{#1}{{\@currentlabel}{\thepage}%
+          {\@currentlabelname}{\@currentHref}{\@kernel@currentdata}}}%
```

Comment:
is a full expansion (or any expansion for that matter) really desired for the fifth argument, or shouldn't tjat be passed unchanges (after one expansion through the .aux mechanism?

Reply:
Agreed: *full* expansion may be very dangerous, but one-level seems to be necessary.